src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -0,0 +1,228 @@
+        for (RemoteAgentFactory factory : Jenkins.getActiveInstance().getExtensionList(RemoteAgentFactory.class)) {
+            if (factory.isSupported(launcher, listener)) {
+                try {
+                    listener.getLogger().println("[ssh-agent]   " + factory.getDisplayName());
+                    agent = factory.start(launcher, listener);
+                    break;
+                } catch (Throwable t) {
+                    faults.put(factory.getDisplayName(), t);
+                }
+            }
+        }
+        if (agent == null) {
+            listener.getLogger().println("[ssh-agent] FATAL: Could not find a suitable ssh-agent provider");
+            listener.getLogger().println("[ssh-agent] Diagnostic report");
+            for (Map.Entry<String, Throwable> fault : faults.entrySet()) {
+                listener.getLogger().println("[ssh-agent] * " + fault.getKey());
+                StringWriter sw = new StringWriter();
+                fault.getValue().printStackTrace(new PrintWriter(sw));
+                for (String line : StringUtils.split(sw.toString(), "\n")) {
+                    listener.getLogger().println("[ssh-agent]     " + line);
+                }
+            }
+            throw new RuntimeException("[ssh-agent] Could not find a suitable ssh-agent provider.");
+        }
+
+        for (SSHUserPrivateKey userPrivateKey : userPrivateKeys) {
+            final Secret passphrase = userPrivateKey.getPassphrase();
+            final String effectivePassphrase = passphrase == null ? null : passphrase.getPlainText();
+            for (String privateKey : userPrivateKey.getPrivateKeys()) {
+                agent.addIdentity(privateKey, effectivePassphrase, SSHAgentBuildWrapper.description(userPrivateKey));
+            }
+        }
+
+        listener.getLogger().println(Messages.SSHAgentBuildWrapper_Started());
+        socket = agent.getSocket();
+        sockets.add(socket);
+    }
+
+    /**
+     * Shuts down the current SSH Agent and purges socket files.
+     */
+    private void cleanUp() {
+        try {
+            TaskListener listener = getContext().get(TaskListener.class);
+            if (agent != null) {
+                agent.stop();
+                listener.getLogger().println(Messages.SSHAgentBuildWrapper_Stopped());
+            }
+        } catch (Throwable th) {
+            getContext().onFailure(th);
+        }
+        purgeSockets();
+    }
+
+    /**
+     * Purges all socket files created previously.
+     * Especially useful when Jenkins is restarted during the execution of this step.
+     */
+    private void purgeSockets() {
+        Iterator<String> it = sockets.iterator();
+        while (it.hasNext()) {
+            File socket = new File(it.next());
+            if (socket.exists()) {
+                if (!socket.delete()) {
+                    listener.getLogger().format("It was a problem removing this socket file %s", socket.getAbsolutePath());
```

Comment:
Did you want to `continue` here, to match the previous behavior of this method (leaving the socket in the list)?